### PR TITLE
batching for historical property data requests

### PR DESCRIPTION
**Pull request by boweihan**
## Overview
Batch requests by frame for historical property data

Screen Shot 2022-05-16 at 3 37 28 PM

Implementation notes:
- batches are sliced by maxResults, mirroring the [API behaviour](https://docs.aws.amazon.com/iot-sitewise/latest/APIReference/API_BatchGetAssetPropertyValueHistory.html#iotsitewise-BatchGetAssetPropertyValueHistory-request-entries).
- success and error handling should remain the same everywhere except in the client layer.
- batch timeframe is not finalized, but currently per browser frame.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -64,7 +64,8 @@
     "cypress-wait-until": "^1.7.2",
     "jest": "26.3.0",
     "jest-cli": "^26.5.1",
-    "vue": "^3.2.26"
+    "vue": "^3.2.26",
+    "uuid": "^3.3.2"
```
Comment (boweihan) [inline packages/components/package.json:68]: This fixes a build issue.

File: packages/components/src/integration/iot-kpi/iot-kpi.spec.component.ts [hunk 2]
```diff
@@ -13,6 +16,15 @@ describe('kpi', () => {
   const assetModelId = 'some-asset-model-id';
 
   before(() => {
+    cy.intercept('/properties/batch/history', (req) => {
```
Comment (boweihan) [inline packages/components/src/integration/iot-kpi/iot-kpi.spec.component.ts:19]: These requests were already being made but the errors ignored

File: packages/source-iotsitewise/src/time-series-data/client/batchGetHistoricalPropertyDataPoints.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                // use 2D array indices as entryIDs to guarantee uniqueness
+                // entryId is used to map batch entries with the appropriate callback
+                const entryId = String(`${requestIndex}-${entryIndex}`);
+
+                // save request entry data in functional closure.
+                callbackCache[entryId] = {
+                  onError: ({ errorMessage: msg = 'batch error', errorCode: status }) => {
+                    onError({
+                      id,
+                      resolution: 0,
```
Comment (diehbria) [inline packages/source-iotsitewise/src/time-series-data/client/batchGetHistoricalPropertyDataPoints.ts:65]: is it correct to always report the raw data as what's failing?

Comment (diehbria) [inline packages/source-iotsitewise/src/time-series-data/client/batchGetHistoricalPropertyDataPoints.ts:65]: always reported as raw data?

Comment (boweihan) [reply packages/source-iotsitewise/src/time-series-data/client/batchGetHistoricalPropertyDataPoints.ts:65]: yeah, this mirrors the existing behaviour - https://github.com/awslabs/iot-app-kit/blob/main/packages/source-iotsitewise/src/time-series-data/client/getHistoricalPropertyDataPoints.ts#L78.

this particular path only handles raw data, there will be a similar method for handling aggregated data coming in a future PR

**Review by diehbria — APPROVED** [review]

**Review by NorbertNader — APPROVED** [review]

**Pull request merged**